pkg/cmd/cli/cmd/edit.go

```diff
@@ -59,14 +78,24 @@ saved copy to include the latest resource version.`
 
 // NewCmdEdit implements the OpenShift cli edit command
 func NewCmdEdit(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
-	filenames := []string{}
+	var options EditOptions
+	options.fullName = fullName
 	cmd := &cobra.Command{
 		Use:     "edit (RESOURCE/NAME | -f FILENAME)",
 		Short:   "Edit a resource on the server",
 		Long:    editLong,
 		Example: fmt.Sprintf(editExample, fullName),
 		Run: func(cmd *cobra.Command, args []string) {
-			err := RunEdit(fullName, f, out, cmd, args, filenames)
+			options.cmd = cmd
+			options.args = args
+
+			if err := options.Complete(); err != nil {
+				cmdutil.CheckErr(err)
+			}
+			if err := options.Validate(); err != nil {
+				cmdutil.CheckErr(err)
+			}
+			err := options.RunEdit()
```

Comment:
@romanbartl to keep the style consistent, you could change this part to be similar to [deploy.go](https://github.com/openshift/origin/blob/255307e708defdbdb66fc421cc3d45cce375a209/pkg/cmd/cli/cmd/deploy.go#L93-L103), using the first style mentioned above.